.github/workflows/python-versions.yml

```diff
@@ -62,7 +62,8 @@ jobs:
   py_snowflake:
     runs-on: ubuntu-latest
 
-    if: github.repository == 'streamlit/streamlit'
+    # Won't run if PR from fork, since that PR wouldn't have secrets access
+    if: github.event.pull_request.head.repo.full_name == github.repository
 
```

Comment:
I think this `if` was intended to check for forks, but please confirm if this was not the intended behavior.

Reply:
I'm not sure if this will work since `github.event.pull_request.head.repo.full_name` may not have a value when this workflow runs on `develop`.

I do think that the intention of this is to prevent this workflow from being run on PRs made from forks, though (@sfc-gh-kbregula can confirm), so we'll want to do this once the problem above is addressed

Reply:
Yikes didn't think of that - I'll revise!

Reply:
Hm, I guess the concern now is whether the first check will run on the `develop` branch of a non-streamlit repo

This is surprisingly hard to get right, and I blame GitHub actions for not supporting what should be a very common thing to do in the OS world 😑

Reply:
Yeah this is unfortunately tricky & currently blocking nightlies since its part of required tests 😓 
Also, this python test workflow does run on the develop branch of a non-streamlit repos as of now...

Reply:
What I wanted here is to block code from running that isn't in the streamlit/streamlit repository because it's untrusted (unsafe), so it can't access the secret.